### PR TITLE
create recipe for mlsauce

### PR DESCRIPTION
**Pull request by thierrymoudiki**
Checklist
- [ x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ x] Source is from official source.
- [ x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ x] If static libraries are linked in, the license of the static library is packaged.
- [ x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ x] Build number is 0.
- [x ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thierrymoudiki** [comment]
Hi, is there something missing in this PR?

**Comment by thierrymoudiki** [comment]
@conda-forge/core Anything new on this?

**Comment by thierrymoudiki** [comment]
@conda-forge/staged-recipes  ready for review

**Comment by thierrymoudiki** [comment]
@conda-forge/staged-recipes ready for review

File: recipes/mlsauce/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    - pandas >=1.3.5
+    - jax     # [unix]
+    - jaxlib      # [unix]
+    - joblib >=0.14.0
+    - six >=1.16.0
+    - threadpoolctl >=2.0.0
+    - tqdm >=4.48.1
+    - {{ pin_compatible('numpy') }}
+
+test:
```
Comment (xhochy) [inline recipes/mlsauce/meta.yaml:52]:
```suggestion
test:
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/mlsauce/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - setuptools
+    - Cython >=0.29.21
```
Comment (xhochy) [inline recipes/mlsauce/meta.yaml:26]: In this section you should only list the packages that you need to build the package. Probably most of the dependencies are run-time only.

File: recipes/mlsauce/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    - matplotlib-base
+    - numpy >=1.14.0
```
Comment (xhochy) [inline recipes/mlsauce/meta.yaml:28]: Always list numpy without a constraint in host if you compile something
```suggestion
    - numpy
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by thierrymoudiki — COMMENTED** [review]
Review done. 
Thanks

**Comment by thierrymoudiki** [comment]
@conda-forge/staged-recipes ready for review

**Comment by xhochy** [comment]
> @xhochy any chance that this could be automated someday?

Update of dependencies? Add this to the `conda-forge.yml` in the feedstock:

```
bot:
  inspect: update-grayskull
```

File: recipes/mlsauce/meta.yaml (hunk 2 above)
Comment (xhochy) [reply recipes/mlsauce/meta.yaml:26]: I would guess that you need to remove more the sections here. Potentially you could use grayskull to regenerate this recipe. It should give you a correct separation of host and run dependencies.

**Comment by thierrymoudiki** [comment]
> > @xhochy any chance that this could be automated someday?
> 
> Update of dependencies? Add this to the `conda-forge.yml` in the feedstock:
> 
> ```
> bot:
>   inspect: update-grayskull
> ```

Done

File: recipes/mlsauce/meta.yaml (hunk 2 above)
Comment (thierrymoudiki) [reply recipes/mlsauce/meta.yaml]: @xhochy done

**Comment by thierrymoudiki** [comment]
@xhochy Done. So, it's not enough to have all the green checks?

File: recipes/mlsauce/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - threadpoolctl >=2.0.0
+    - tqdm >=4.48.1
+    - {{ pin_compatible('numpy') }}
+
+test:
+  requires:
+    - pip
+  commands:
+    - pip check
+  imports: mlsauce
```
Comment (hmaarrfk) [inline recipes/mlsauce/meta.yaml:49]:
```suggestion
  imports: 
    - mlsauce
```

File: recipes/mlsauce/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/Techtonique/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: a0b0120941aa04173d85e412e4857889ed22906bd9dcc4bec46d694883144fd3
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [py < 35]
```
Comment (hmaarrfk) [inline recipes/mlsauce/meta.yaml:15]:
```suggestion
```
this is not needed as we build only for python 3.8+ at conda forge today.

File: recipes/mlsauce/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  host:  
+    - Cython >=0.29.21  
+    - numpy
+    - pip
+    - python    
+    - setuptools
+    - scipy    
+  run:
+    - python
+    - Cython >=0.29.21
```
[the remark this answers is not in the document]
Comment (thierrymoudiki) [reply recipes/mlsauce/meta.yaml:30]: I'm checking whether it works with this requirement or not

**Comment by thierrymoudiki** [comment]
@hmaarrfk how about this new version?

**Review by xhochy — APPROVED** [review]
Looks good now

**Pull request merged**